Modules/Filtering/ImageFrequency/include/itkUnaryFrequencyDomainFilter.h

```diff
@@ -0,0 +1,202 @@
+/*=========================================================================
+ *
+ *  Copyright Insight Software Consortium
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *=========================================================================*/
+#ifndef itkUnaryFrequencyDomainFilter_h
+#define itkUnaryFrequencyDomainFilter_h
+
+#include <itkInPlaceImageFilter.h>
+#include <itkFrequencyFFTLayoutImageRegionIteratorWithIndex.h>
+
+namespace itk
+{
+/** \class UnaryFrequencyDomainFilter
+ * \brief Performs a unary operation on a frequency domain image
+ *
+ * A frequency filtering functor needs to be supplied via one of SetFunctor() overloads.
+ * The functor should take FrequencyIteratorType reference as its only parameter.
+ * If functor configurability is required,
+ * those parameters should be passed directly to the functor at the place of definition.
+ *
+ *
+ * Filters in the module ITKImageFrequency work with input images in the frequency domain.
+ * This filter is templated over a TFrequencyIterator depending on the
+ * frequency layout of the input image.
+ * Images in the dual space can be acquired experimentally, from scattering exaperiments or other techniques.
+ * The layout of these images is the same as spatial domain images. Use \ref FrequencyImageRegionIteratorWithIndex
+ *
+ * Frequency-domain images can be computed from any spatial-domain applying a Fourier Transform \ref ForwardFFTImageFilter.
+ * Use \ref FrequencyFFTLayoutImageRegionIteratorWithIndex.
```

Comment:
Use -> If \ref ForwardFFTImageFilter was used for the domain transformation, template this filter with the \ref FrequencyFFTLayoutImageRegionIteratorWithIndex.

Reply:
> Note: Should we then enumerate here all the possibilities? FFTHalfHermitian, FFTShifted?

:+1: good idea.